libs/sdk-core/src/breez_services.rs

```diff
@@ -453,34 +453,42 @@ impl BreezServices {
         Ok(None)
     }
 
+    /// See [ReverseSwapperAPI::reverse_swap_pair_info]
     pub async fn reverse_swap_info(&self) -> Result<ReverseSwapPairInfo> {
-        self.btc_send_swapper.reverse_swap_info().await
+        self.btc_send_swapper.reverse_swap_pair_info().await
     }
 
-    /// The steps for a full reverse swap are:
-    ///
-    /// 1. User starts reverse swap (input: desired amount, destination BTC address)
-    /// 2. Receive HODL invoice from Boltz
-    /// 3. Pay the HODL invoice
-    /// 4. Monitor on-chain "lock" tx (poll), notify user on confirmation
-    /// 5. Broadcast "claim" tx (reveal preimage)
-    ///
-    /// This method covers steps 1 and 2 above.
+    /// Creates a reverse swap and attempts to pay the HODL invoice
     pub async fn send_onchain(
         &self,
         amount_sat: u64,
         onchain_recipient_address: String,
         pair_hash: String,
     ) -> Result<ReverseSwapInfo> {
-        let routing_hop = self.lsp_info().await?;
-        self.btc_send_swapper
-            .create_reverse_swap(
-                amount_sat,
-                onchain_recipient_address.clone(),
-                pair_hash,
-                routing_hop.pubkey,
-            )
-            .await
+        match self.in_progress_reverse_swap().await? {
+            None => self.btc_send_swapper
+                .create_reverse_swap(
+                    amount_sat,
+                    onchain_recipient_address,
+                    pair_hash,
+                    self.lsp_info().await?.pubkey,
+                )
+                .await,
+            Some(_) => Err(anyhow!(
+                "There already is a Reverse Swap in progress, you can only start a new one after this one finishes. \
+                Use the in_progress_reverse_swap method to get the current swap's state."
+            ))
+        }
+    }
+
+    /// Returns an optional in-progress [ReverseSwapInfo]
+    pub async fn in_progress_reverse_swap(&self) -> Result<Option<ReverseSwapInfo>> {
+        let monitored = self.btc_send_swapper.list_monitored().await?;
+        match monitored.len() {
+            x if x == 0 => Ok(None),
+            x if x == 1 => Ok(monitored.first().cloned()),
+            _ => Err(anyhow!("Found more than one ongoing reverse swaps")),
```

Comment:
We don't need to return error here. In theory there could be more than one in progress swap for example when running the node in multiple apps. IMO we can either return the first one or return a list of swaps.

Reply:
Good point, addressed it in 5c0376deebe8e3d3aef9c1c178612aa45a624041

This also answers an open question I added earlier today about exactly this
> Clarify meaning of "in progress rev swap": does it covered all monitored stauses? if not (e.g. ClaimTxMempool = in progress), then the API call to `list_in_progress_rev_swaps` can theoretically return more than one?